### PR TITLE
refactor: avoid git fetch on each build

### PR DESCRIPTION
**Pull request by hferentschik**
- per default VERSION will be set to a dev version
- removed if statement checking for VERSION file
- introduced target set-release-version to make sure the latest tag is used to
  determine next version (using git fetch)
- extracted Docker related targets into Makefile.docker

fixes issue #3490

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -14,16 +14,17 @@
 # limitations under the License.
 #
 
+# Make does not offer a recursive wildcard function, so here's one:
+rwildcard=$(wildcard $1$2) $(foreach d,$(wildcard $1*),$(call rwildcard,$d/,$2))
```
Comment (hferentschik) [inline Makefile]: @jstrachan - I double checked that his works for macOS. I am using 'GNU Make 3.81'. If it does not work for you, I think we can safely remove it. There is really no big benefit here. We can rely on Go figuring things out.

File: Makefile [hunk 2]
```diff
@@ -32,18 +33,9 @@ GITHUB_ACCESS_TOKEN := $(shell cat /builder/home/git-token 2> /dev/null)
 PEGOMOCK_PACKAGE := github.com/petergtz/pegomock/
 CGO_ENABLED = 0
 
-all: build
-full: check
-check: build test
+TAG := $(shell git describe --abbrev=0 --tags 2>/dev/null)
+VERSION := $(shell echo "$(TAG)-dev+$(REV)" | sed 's/^v//')
```
Comment (hferentschik) [inline Makefile]: @jstrachan per default we create a dev version without `git fetch`

File: Makefile [hunk 3]
```diff
@@ -159,30 +141,40 @@ inttestbin:
...
 
-release: check
+set-release-version:
+    # ensure we have the latest tag
+	@git fetch --all -q 2>/dev/null
+	$(eval VERSION = $(shell git describe --abbrev=0 --tags 2>/dev/null | sed 's/^v//' ))
+	@echo Release version: $(VERSION)
+
+.PHONY: release
+release: set-release-version check
```
Comment (hferentschik) [inline Makefile]: @jstrachan - for the release we depend on a new `set-release-version` which calls `git fetch` and dynamically updates VERSION using eval.

**Comment by jstrachan** [comment]
@hferentschik btw when we are creating a new release there is the `$VERSION` env var we could use to simplify the makefile?

**Comment by hferentschik** [comment]
> btw when we are creating a new release there is the $VERSION env var we could use to simplify the makefile?

Ahh, even better. In this case, we could do something like create a dev VERSION if there is none set, otherwise use the VERSION as passed through the environment. Makes much more sense.

**Comment by jstrachan** [comment]
@hferentschik yeah - then folk could set a dev `$VERSION` env var once on startup then we never have  to run any git commands on a build which will further speed things up a bit

File: Makefile [hunk 4]
```diff
@@ -32,18 +33,9 @@ GITHUB_ACCESS_TOKEN := $(shell cat /builder/home/git-token 2> /dev/null)
 PEGOMOCK_PACKAGE := github.com/petergtz/pegomock/
 CGO_ENABLED = 0
 
-all: build
-full: check
-check: build test
+# set dev version unless VERSION is explicitly set via environment
+VERSION ?= $(shell echo "$$(git describe --abbrev=0 --tags 2>/dev/null)-dev+$(REV)" | sed 's/^v//')
```
Comment (hferentschik) [inline Makefile]: @jstrachan, only set VERSION if not explicitly specified via the environment. For the case VERSION is not set use `git describe`.

**Comment by hferentschik** [comment]
@jstrachan 

> yeah - then folk could set a dev $VERSION env var once on startup then we never have to run any git commands on a build which will further speed things up a bit

+1 I updated the Makefile. If VERSION is set in the environment use it, otherwise `git describe` is used. Without the fetch it is not really a problem. Also, if someone is bothered by it, he can always set now an explicit VERSION.

**Comment by hferentschik** [comment]
@jstrachan on a different note, could `Jenkinsfile` and myvalues.yaml.template` not be removed now?

**Comment by jstrachan** [comment]
@hferentschik we can zap `Jenksinfile` now yeah - though we still use `myvalues.yaml.template` to run the BDD tests

**Pull request merged**